pkg/controller/controller_binding.go

```diff
@@ -348,39 +351,70 @@ func isPlanBindable(serviceClass *v1alpha1.ServiceClass, plan *v1alpha1.ServiceP
 }
 
 func (c *controller) injectBinding(binding *v1alpha1.Binding, credentials map[string]interface{}) error {
-	glog.V(5).Infof("Creating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
-	secret := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      binding.Spec.SecretName,
-			Namespace: binding.Namespace,
-		},
-		Data: make(map[string][]byte),
-	}
+	glog.V(5).Infof("Creating/updating Secret %v/%v", binding.Namespace, binding.Spec.SecretName)
 
+	secretData := make(map[string][]byte)
 	for k, v := range credentials {
 		var err error
-		secret.Data[k], err = serialize(v)
+		secretData[k], err = serialize(v)
 		if err != nil {
+			// Terminal error
+			// TODO mark as terminal error once we have the terminal condition
 			return fmt.Errorf("Unable to serialize credential value %q: %v; %s",
 				k, v, err)
 		}
 	}
 
-	found := false
-
-	_, err := c.kubeClient.Core().Secrets(binding.Namespace).Get(binding.Spec.SecretName, metav1.GetOptions{})
+	// Creating/updating the Secret
+	secretClient := c.kubeClient.Core().Secrets(binding.Namespace)
+	existingSecret, err := secretClient.Get(binding.Spec.SecretName, metav1.GetOptions{})
 	if err == nil {
-		found = true
+		// Update existing secret
+		if !IsControlledBy(existingSecret, binding) {
+			controllerRef := GetControllerOf(existingSecret)
+			// TODO mark as terminal error once we have the terminal condition
+			return fmt.Errorf("Secret '%s' is not owned by Binding, controllerRef: %v",
+				existingSecret.Name, controllerRef)
+		}
+		existingSecret.Data = secretData
+		_, err = secretClient.Update(existingSecret)
```

Comment:
The way these if-statements are structured scares me.  If I'm just being paranoid don't hesitate to say so :-) but I've seen cases where people accidentally add, or leave off, a colon on assignments and they end up checking the wrong variable at times.  In this case the 'err' on the previous line is the same 'err' that's defined outside of this if-block.  And everything you have here is correct.  However, the next time someone comes in here and twiddles the code they might (by mistake) add a ":" as they're re-working things and create a new 'err' variable that's scoped to this inner "if".  Meaning, the IF on line 410 below will be checking a different 'err' and we'll end up treating an error case as a success.  And I question if our tests will catch it.  This just feels like one of those hard to track-down/intermittent bugs waiting to happen.

Can I suggest that we either duplicate the "if" from 410 in here so we're sure we're using the correct "err" each time (and do it in the "else" side), or if we're going to use a func-scope variable then we rename it something else since "err" is just too commonly used and the odds of someone adding an inner-scoped version of it by mistake will be very slim?